File: LeetCode/780. Reaching Points/Solution.py
class Solution:
    def reachingPoints(self, sx: int, sy: int, tx: int, ty: int) -> bool:

        while tx >= sx and ty >= sy:
            if ty > tx:
                if tx > sx:
                    ty %= tx
                elif tx == sx:
                    return (ty - sy) % tx == 0
            elif tx > ty:
                if ty > sy:
                    tx %= ty
                elif ty == sy:
                    return (tx - sx) % ty == 0
            else:
                break

        return (ty == sy and tx == sx)

File: LeetCode/780. Reaching Points/Solution.dart
class Solution {
  bool reachingPoints(int sx, int sy, int tx, int ty) {
    while (tx >= sx && ty >= sy) {
      if (tx > ty) {
        if (ty > sy) {
          tx %= ty;
        } else {
          return (tx - sx) % ty == 0;
        }
      } else if (ty > tx) {
        if (tx > sx) {
          ty %= tx;
        } else {
          return (ty - sy) % tx == 0;
        }
      } else {
        break;
      }
    }
    return (tx == sx && ty == sy);
  }
}
